src/sage/combinat/root_system/cartan_matrix.py

```diff
@@ -1135,13 +1135,24 @@ def find_cartan_type_from_matrix(CM):
         ['C', 3] relabelled by {1: 0, 2: 1, 3: 2}
         sage: CM = CartanMatrix([[2,-1,-2], [-1,2,-1], [-2,-1,2]])
         sage: find_cartan_type_from_matrix(CM)
+
+    TESTS:
+
+    Check that :issue:`35987` is fixed::
+
+        sage: cm = CartanMatrix(['A',7]).subtype([2,3,5])
+        sage: from sage.combinat.root_system.cartan_matrix import find_cartan_type_from_matrix
+        sage: find_cartan_type_from_matrix(cm)
+        A2xA1 relabelled by {1: 2, 2: 3, 3: 5}
```

Comment:
Would be nice to add an example with type B.

Reply:
I did one better and did affine type B cut into 3 parts.